.github/workflows/build-git-installers.yml

```diff
@@ -302,88 +302,259 @@ jobs:
+          name: tmp.osx-pkg
+          path: |
+            git/.github/disk-image
+
+  osx_sign_and_notarize_pkg:
+    # ESRP service requires signing to run on Windows
+    runs-on: windows-latest
+    needs: osx_pack
+    steps:
+    - name: Check out repository
+      uses: actions/checkout@v3
+      with:
+          path: 'git'
+
+    - name: Download unsigned package
+      uses: actions/download-artifact@v3
+      with:
+        name: tmp.osx-pkg
+        path: pkg
+
+    - name: Zip unsigned package
+      shell: pwsh
+      run: |
+        Compress-Archive -Path pkg/*.pkg pkg/msft-git-pkg.zip
+        cd pkg
+        Get-ChildItem -Exclude msft-git-pkg.zip | Remove-Item -Recurse -Force
+
+    - uses: azure/login@v1
+      with:
+        creds: ${{ secrets.AZURE_CREDENTIALS }}
+
+    - name: Set up ESRP client
+      shell: pwsh
+      env:
+        AZURE_STORAGE_KEY: ${{ secrets.AZURE_STORAGE_KEY }}
+        AZURE_VAULT: ${{ secrets.AZURE_VAULT }}
+        AUTH_CERT: ${{ secrets.AZURE_VAULT_AUTH_CERT_NAME }}
+        REQUEST_SIGNING_CERT: ${{ secrets.AZURE_VAULT_REQUEST_SIGNING_CERT_NAME }}
+      run: |
+        git\.github\scripts\set-up-esrp.ps1
+
+    - name: Sign package
+      shell: pwsh
+      env:
+        AZURE_AAD_ID: ${{ secrets.AZURE_AAD_ID }}
+        # We temporarily need two AAD IDs, as we're using an SSL certificate associated
+        # with an older App Registration until we have the required hardware to approve
+        # the new certificate in SSL Admin.
+        AZURE_AAD_ID_SSL: ${{ secrets.AZURE_AAD_ID_SSL }}
+        APPLE_KEY_CODE: ${{ secrets.APPLE_KEY_CODE }}
+        APPLE_SIGNING_OP_CODE: ${{ secrets.APPLE_SIGNING_OPERATION_CODE }}
+      run: |
+        python git\.github\scripts\run-esrp-signing.py pkg $env:APPLE_KEY_CODE $env:APPLE_SIGNING_OP_CODE
+
+    - name: Unzip signed package
+      shell: pwsh
+      run: |
+        mkdir unsigned
+        Expand-Archive -LiteralPath signed\msft-git-pkg.zip -DestinationPath .\unsigned -Force
+        Remove-Item signed\msft-git-pkg.zip -Force
+
+    - name: Notarize signed package
+      shell: pwsh
+      env:
+        AZURE_AAD_ID: ${{ secrets.AZURE_AAD_ID }}
+        # We temporarily need two AAD IDs, as we're using an SSL certificate associated
+        # with an older App Registration until we have the required hardware to approve
+        # the new certificate in SSL Admin.
+        AZURE_AAD_ID_SSL: ${{ secrets.AZURE_AAD_ID_SSL }}
+        APPLE_KEY_CODE: ${{ secrets.APPLE_KEY_CODE }}
+        APPLE_NOTARIZATION_OP_CODE: ${{ secrets.APPLE_NOTARIZATION_OPERATION_CODE }}
+      run: |
+        python git\.github\scripts\run-esrp-signing.py unsigned $env:APPLE_KEY_CODE `
+          $env:APPLE_NOTARIZATION_OP_CODE --params 'BundleId' 'com.microsoft.git'
+
+    - name: Upload signed and notarized pkg
+      uses: actions/upload-artifact@v3
+      with:
+        name: osx-signed-pkg
+        path: |
+          signed
+
+  osx_publish_dmg:
```

Comment:
The changes to this `.yml` file make sense to what we had discussed:

1. Build Git.
2. Sign what Git built.
3. Package the signed bits into the `.pkg` and `.dmg`.
4. Sign the `.pkg` and `.dmg`.
5. Publish that `.pkg` and `.dmg`.

Reply:
To be clear, we're not currently signing the `.dmg`. Since this is basically just a "virtual disk" that serves as a vehicle for getting our `.pkg` to users, I don't _think_ we need to worry about signing it. This assumption is supported by the fact that gatekeeper doesn't complain about it not being signed when downloading/installing (caveat: I've done some monkeying with gatekeeper on my machine during this process, so let me know if your experience was different!).

Reply:
That makes sense. Ignore my blathering about the `.dmg`.

It was so nice to just double-click the `.pkg` and install without the hoopla about warnings and right-click "open" stuff.